src/components/ActionLink/ActionLink.vue

```diff
@@ -0,0 +1,161 @@
+<!--
+  - @copyright Copyright (c) 2019 John Molakvoæ <skjnldsv@protonmail.com>
+  -
+  - @author John Molakvoæ <skjnldsv@protonmail.com>
+  -
+  - @license GNU AGPL version 3 or any later version
+  -
+  - This program is free software: you can redistribute it and/or modify
+  - it under the terms of the GNU Affero General Public License as
+  - published by the Free Software Foundation, either version 3 of the
+  - License, or (at your option) any later version.
+  -
+  - This program is distributed in the hope that it will be useful,
+  - but WITHOUT ANY WARRANTY; without even the implied warranty of
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+  -
+  -->
+
+<template>
+	<a :href="href"
+		class="action-link focusable"
+		rel="noreferrer noopener">
+
+		<!-- icon -->
+		<span :class="[isIconUrl ? 'action-link__icon--url' : icon]"
+			:style="{ backgroundImage: isIconUrl ? `url(${icon})` : null }"
+			class="action-link__icon" />
+
+		<!-- long text with title -->
+		<p v-if="title">
+			<strong class="action-link__title">
+				{{ title }}
+			</strong>
+			<br>
+			<!-- white space is shown on longtext, so we can't
+				put {{ text }} on a new line for code readability -->
+			<span class="action-link__longtext">{{ text }}</span>
+		</p>
+
+		<!-- long text only -->
+		<!-- white space is shown on longtext, so we can't
+			put {{ text }} on a new line for code readability -->
+		<p v-else-if="isLongText" class="action-link__longtext">{{ text }}</p>
+
+		<!-- default text display -->
+		<span v-else class="action-link__text">{{ text }}</span>
+
+		<!-- fake slot to gather inner text -->
+		<slot v-if="false" />
+	</a>
+</template>
+
+<script>
+export default {
+	name: 'ActionLink',
+
+	props: {
+		href: {
+			type: String,
+			default: '#',
+			required: false
```

Comment:
I would argue a link without href does not make sense. This should be required.

Reply:
No, because the default is `#`, so this is not required :)

Reply:
how does an `ActionLink` make sense if you don't specify a link? What use case would that cover?

Reply:
(the default should be removed as well)

Reply:
@ChristophWurst if there is no action nor link, so just text I think?
For only action I guess a button is better? No idea really :man_shrugging:

Reply:
But an `a` to `#` has a certain semantic. This might be problematic for accessibility and if the click event is not prevented the browser changes the URL and therefore triggers a Vue router navigation.

My suggestion:
* If you want to render an `a` with an external link: ActionLink
* If you want to render a button with a js callback: ActionButton
* If you want to display text only: ActionText
* If you want to render a Vue router link: ActionRouterLink

Reply:
@ChristophWurst shall we really have empty text without actions? (ActionText)
And Router in popover as well? ^^

Reply:
Yes, I use a text-only entry in Mail to display info about a folder. I think this is useful.

And vue router makes sense as well, e.g. for the *Edit* action of a Mail account. Having a router link makes it possible to open that link in a new tab. With callback and programmatic navigation this does not work.

Reply:
Right now I even have to disable an eslint rule that complains that I'm using a sideeffect inside a computed property -> the programmatic navigation inside the callback of the *Edit* action of an account.